test/torchaudio_unittest/datasets/speechcommands_test.py

```diff
@@ -53,43 +53,59 @@ class TestSpeechCommands(TempDirMixin, TorchaudioTestCase):
         seed = 0
-        for label in LABELS:
-            path = os.path.join(dataset_dir, label)
-            os.makedirs(path, exist_ok=True)
-            for j in range(2):
-                # generate hash ID for speaker
-                speaker = "{:08x}".format(j)
-
-                for utterance in range(3):
-                    filename = f"{speaker}{speechcommands.HASH_DIVIDER}{utterance}.wav"
-                    file_path = os.path.join(path, filename)
-                    seed += 1
-                    data = get_whitenoise(
-                        sample_rate=sample_rate,
-                        duration=0.01,
-                        n_channels=1,
-                        dtype="int16",
-                        seed=seed,
-                    )
-                    save_wav(file_path, data, sample_rate)
-                    sample = (
-                        normalize_wav(data),
-                        sample_rate,
-                        label,
-                        speaker,
-                        utterance,
-                    )
-                    cls.samples.append(sample)
+        valid_file = os.path.join(dataset_dir, "validation_list.txt")
+        test_file = os.path.join(dataset_dir, "testing_list.txt")
+        with open(valid_file, "w") as valid, open(test_file, "w") as test:
+            for label in LABELS:
+                path = os.path.join(dataset_dir, label)
+                os.makedirs(path, exist_ok=True)
+                for j in range(6):
+                    # generate hash ID for speaker
+                    speaker = "{:08x}".format(j)
+
+                    for utterance in range(3):
+                        filename = f"{speaker}{speechcommands.HASH_DIVIDER}{utterance}.wav"
+                        file_path = os.path.join(path, filename)
+                        seed += 1
+                        data = get_whitenoise(
+                            sample_rate=sample_rate,
+                            duration=0.01,
+                            n_channels=1,
+                            dtype="int16",
+                            seed=seed,
+                        )
+                        save_wav(file_path, data, sample_rate)
+                        sample = (
+                            normalize_wav(data),
+                            sample_rate,
+                            label,
+                            speaker,
+                            utterance,
+                        )
+                        cls.samples.append(sample)
+                        if j < 2:
+                            cls.train_samples.append(sample)
+                        elif j < 4:
+                            label_filename = os.path.join(label, filename)
+                            valid.write(f'{label_filename}\n')
+                            cls.valid_samples.append(sample)
+                        elif j < 6:
+                            label_filename = os.path.join(label, filename)
```

Comment:
The contents are of `"testing_list.txt"` and `"validation_list.txt"` are in Linux format. 

```
$ head speech_commands_v0.01/testing_list.txt
bed/0c40e715_nohash_0.wav
bed/0ea0e2f4_nohash_0.wav
bed/0ea0e2f4_nohash_1.wav
```

Here, using `os.path.join` make it to the valid path on Windows system (something like `bed\0c40e715_nohash_0.wav`), so this is not correct. Something like the following is the correct way to mimic the data.

```suggestion
                            label_filename = f'{label}/{filename}'
```

Same goes to the case where `j < 4`

Reply:
good catch

Reply:
I think it's more straightforward to use `List[pathlib.Path]` objects for `self._walker`, rather than converting them to `List[str]`.

Reply:
Yeah, that will be more convenient to do that in the future